### ISSUE TITLE
a11y: Missing instruction for mandatory fields (LabeledSelect and LabeledInput)

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#14255]
**Setup**
- Rancher version:
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**

`a11y issue 99`

Although an asterisk (*) is provided for the mandatory fields in "Change Password" form, the instruction stating the meaning of the asterisk (*) is not provided resulting in cognitive and screen reader users not be able to identify the mandatory fields.

Change Password 
- Current Password input field
- New Password input field
- Confirm Password input field

Pod Security Admission: Create
- Name input field

Repository: Create
- Name input field
- Index URL (available after selecting "http(s) URL to an index generated by Helm" radio button)
- Git Repo URL (available after selecting "Git repository containing Helm chart or cluster template definitions" radio button)
- OCI Repository Host URL (available after selecting "OCI Repository Experimental" radio button)

Git Repo - Add Repository
- Name input field
- Branch Name input field

User: Create
- Username input field
- New Password
- Confirm Password

Users Retention Settings
- User retention process schedule (available on activating Delete user accounts after an inactivity period (duration since last login) checkbox)

**To Reproduce**

**Result**
Although an asterisk (*) is provided for the mandatory fields in "Change Password" form, the instruction stating the meaning of the asterisk (*) is not provided resulting in cognitive and screen reader users not be able to identify the mandatory fields.

**Expected Result**

The meaning of asterisk (*) should be described in plain text so that all users can understand it.

**Screenshots**

**Additional context**

**Comment by yonasberhe23** [comment]
Unit tests added and did a quick manual check - LabeledSelect and LabeledInput components contain `aria-required` attribute in areas I checked on latest Rancher build. 

- Rancher v2.12-b256ba13c96d58bac0ef2071c0a37ced7f1e21d4-head
- Dashboard master `ac02757`
